ARCs/arc-0035.md

```diff
@@ -0,0 +1,119 @@
+
+
+## Abstract
+This document outlines the high-level requirements for a wallet-agnostic backup protocol that can be used across all wallets on the Algorand ecosystem.
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+### Requirements
+At a high-level, offline wallet backup protocol has the following requirements:
+
+ * Wallet applications should allow backing up and storing multiple accounts at the same time.
+Account information should be encrypted with a user defined secret key, utilizing NaCl SecretBox method (audited and endorsed by Algorand).
+ 
+ * Encrypted final string should be easily copyable to be stored digitally.
+When importing, wallet applications should be able to detect already imported accounts and gracefully ignore them.
+
+### Format
+
+Before encryption, account information should be converted to the following JSON format:
+
+```
+{
+  "device_id": "UNIQUE IDENTIFIER FOR DEVICE (OPTIONAL)",
+  "provider_name": "PROVIDER NAME (OPTIONAL, i.e. Pera Wallet)",
+  "accounts": [
+    {
+      "address": "ACCOUNT PUBLIC ADDRESS (REQUIRED)"
+      "name": "USER DEFINED ACCOUNT NAME (OPTIONAL)",
+      "account_type": "TYPE OF ACCOUNT: single, multisig, watch, contact, ledger (REQUIRED)",
+    "private_key": "ACCOUNT PRIVATE KEY AS BASE64 ENCODING OF 64 BYTE PRIVATE KEY as encoded by NACL/libsodium (NOT PASSPHRASE, REQUIRED for user-owned accounts, can be omitted in case of watch, contact, ledger accounts)"
+      "metadata": "ANY ADDITIONAL CONTENT (OPTIONAL)",
+    },
+    ...
+  ],
+}
+```
+
+*Clients must accept additional fields in the JSON document.*
+
+Here is an example with a single account:
+
+```
+{
+  "device_id": "2498232091970170817",
+  "provider_name": "Pera Wallet",
+  "accounts": [
+    {
+      "address": "ELWRE6HZ7KIUT46EQ6PBISGD3ND6QSCBVWICYR2QR2Y7LOBRZRCAIKLWDE"
+      "name": "My NFT Account",
+      "account_type": "single",
+      "private_key": "w0HG2VH7tAYz9PD4SYX0flC4CKh1OONCB6U5bP7cXGci7RJ4+fqRSfPEh54USMPbR+hIQa2QLEdQjrH1uDHMRA=="
+    }
+  ],
+}
+```
+
+### Encryption
+
+Once the input JSON is ready as specified above, it needs to be encrypted. Even if it is assumed that the user is going to store this information in a secure location, copy-pasting it without encryption is not secure since multiple applications can access the clipboard. 
+
+The information needs to be encrypted using a very long passphrase. 12 word mnemonic will be used as the key. 12-word mnemonic is secure and it will not create confusion with the 24-word mnemonics that are used to encrypt accounts.
+
+The wallet applications should not allow users to copy the 12-word mnemonic nor allow taking screenshots. The users should note it visually. 
```

Comment:
After extensive discussion, Pera product team would like this requirement for preventing copy actions removed from ARC-35. What we've seen in the wild is that users will copy text on screen one way or another (users are very resilient against blocks like this). The whole point of ASB is that it makes it safe to copy the 12-word passphrase AND the encrypted backup. That way neither copied item provides direct access to user accounts.

What should be specified here is that the wallets should clearly direct the user to store the 12-word passphrase and the encrypted backup **separately**.